packages/order-experience-api/src/api/getRefund.ts

```diff
@@ -0,0 +1,56 @@
+import {
+  AbstractController,
+  logger,
+  ValidatedRequest,
+} from '@verkkokauppa/core'
+import type { Response } from 'express'
+import * as yup from 'yup'
+import { validateApiKey } from '@verkkokauppa/configuration-backend'
+import { getOrderAdmin, getRefundAdmin } from '@verkkokauppa/order-backend'
+import { getRefundPaymentForOrderAdmin } from '@verkkokauppa/payment-backend'
+const requestSchema = yup.object().shape({
+  params: yup.object().shape({
+    refundId: yup.string().required(),
+  }),
+  headers: yup.object().shape({
+    'api-key': yup.string().required(),
+    namespace: yup.string().required(),
+  }),
+})
+
+export class GetRefundController extends AbstractController<
+  typeof requestSchema
+> {
+  protected readonly requestSchema = requestSchema
+
+  protected async implementation(
+    req: ValidatedRequest<typeof requestSchema>,
+    res: Response
+  ) {
+    const {
+      params: { refundId },
+      headers: { 'api-key': apiKey, namespace },
+    } = req
+    await validateApiKey({ namespace, apiKey })
+
+    const refund = await getRefundAdmin({ refundId })
+
+    const order = await getOrderAdmin(refund.refund)
+
+    let refundPayment = null
+    try {
+      refundPayment = await getRefundPaymentForOrderAdmin({
```

Comment:
Doesn't this possibly pick up refundPayment from another refund made for this order?